lib/karafka/setup/config.rb

```diff
@@ -37,7 +37,7 @@ class Config
         # Connection pool size for producers. Note that we take a bigger number because there
         # are cases when we might have more sidekiq threads than Karafka routes (small app)
         # or the opposite for bigger systems
-        setting :size, -> { [::Karafka::App.routes.count, Sidekiq.options[:concurrency]].max }
+        setting :size, [::Karafka::App.routes.count, Sidekiq.options[:concurrency]].max
```

Comment:
I disagree and I know why - when config is being build dynamically, it is set based on the routes and we know nothing about them in this moment. That's why it is being delegated to be decided later on. This needs to stay as a proc